### PR TITLE
Adding task split run option in data file #78

### PR DESCRIPTION
**Pull request by Slowblitz**

### CODE REVIEW

**Comment by Slowblitz** [comment]
I'll try to fix #79 in this PR too

File: ando/AnDOChecker.py [hunk 1]
```diff
@@ -212,4 +212,6 @@ def main():
 
 
 if __name__ == "__main__":
-    main()
+    test_folder = 'tests/dataset/exp-valid'
```
Comment (JuliaSprenger) [inline ando/AnDOChecker.py:215]: This was only for testing and should not have been commited.

File: ando/rulesStructured.py [hunk 2]
```diff
@@ -41,19 +41,21 @@
 currentdepth_rules['authorized_folders'] = ['^$']
 currentdepth_rules['mandatory_folders'] = []
 currentdepth_rules['authorized_data_files'] = [
-    ['^sub-([a-zA-Z0-9]+)_ses-([a-zA-Z0-9]+)([\\w\\-]*)_ephys', ANY_DATA_EXT]]
+    ['sub-([a-zA-Z0-9]+)_ses-([a-zA-Z0-9]+)([\\w\\-]*)(_task-([a-zA-Z0-9]+))?'
```
Comment (JuliaSprenger) [inline ando/rulesStructured.py:44]: You removed the '^' at the beginning of the regexp patterns, that are anchoring the patterns to the beginning of the strings to match. I don't think this was intended...

File: ando/tests/test_AnDOChecker.py [hunk 3]
```diff
@@ -91,6 +91,10 @@ def test_nonAuthorizedDataFilesError(self):
         path = Path(dir_path) / "dataset" / "exp-nonAuthorizedDataFilesError"
         self.assertEqual(CHK.is_valid(path)[0], False)
 
+    def test_missingDataFile(self):
```
Comment (JuliaSprenger) [inline ando/tests/test_AnDOChecker.py:94]: Maybe it would also be good to check if the expected number of error is correct.

**Review by JuliaSprenger — APPROVED** [review]

**Pull request merged**